samples/log_callback/enclave/enc.c

```diff
@@ -1,13 +1,53 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include <openenclave/tracee.h>
 #include <stdio.h>
+#include <string.h>
 
 // Include the trusted log_callback header that is generated
 // during the build. This file is generated by calling the
 // sdk tool oeedger8r against the log_callback.edl file.
 #include "log_callback_t.h"
 
+void enclave_customized_log(
+    void* context,
+    oe_log_level_t level,
+    uint64_t thread_id,
+    const char* message)
+{
+    char modified_log[200];
+
+    sprintf(
+        modified_log,
+        "E, %s, %llx, %s",
+        oe_log_level_strings[level],
+        thread_id,
+        message);
+
+    /*
+     * Add logic here to modify the log message, to obscure enclave logs from
+     * the host. The context can be used to transfer shared/public/private
+     * secret, that maybe used to encrypt the log message.
+     *
+     * NOTE: fprintf does not work consitently in this function. If used, the
```

Comment:
```suggestion
     * NOTE: fprintf does not work consistently in this function. If used, the
```

Reply:
Is there a bug?  Is there an issue filed?

Reply:
Yes, this is a bug, ( might not be solvable). This bug will be hit only when fprintf (using oehostfs) is called after oe_terminate_enclave() call. I guess logging is the only place where this could happen. I wanted to create an issue and modify the NOTE in code after the sample gets checked-in.

Reply:
Is there an issue filed?  Would be nice to add the issue number into the comment. Alternatively if it is by design (not a bug) then expand comment to explain the issue, like explaining that fprintf intentionally will not work after the oe_terminate_enclave() call.

Reply:
Added issue #4349 to comments in enc.c:enclave_customized_log.